crowbar_framework/app/models/node_object.rb

```diff
@@ -1072,6 +1072,15 @@ def file_exist?(file)
     out[:exit_code].zero?
   end
 
+  def script_finished(script)
+    base = "/var/lib/crowbar/upgrade/" + File.basename(script, ".sh")
+    ok_file = base + "-ok"
+    failed_file = base + "-failed"
+    return "ok" if file_exist? ok_file
+    return "failed" if file_exist? failed_file
```

Comment:
> Well... I don't know. I do not see it as a problem.

Given that this means 100 ssh-connections/second for a 100 compute node setup I am not sure I agree with it :).
I would also be pretty easy get down to half the number, by checking for both files in one command. It also might be an option to just poll a single node and check the next one only after the first finished. (IIRC mkcloud does it this way when waiting for nodes to be ready).

> And I see it as more readable than the alternative threads option.

Note, I didn't say that the Threads approach is the better way here. As it would involves a lot of longer-lived ssh connections it might actually be worse (but we don't actually know).